File: 2 Sorting/B Bubble Sort/bubble-sort.py
def bubble_sort(arr):
  n = len(arr)
  for i in range(n - 1):
    swapped = False
    for j in range(i + 1, n):
      if arr[i] > arr[j]:
        swapped = True
        arr[i], arr[j] = arr[j], arr[i]
    if not swapped:
      break

arr = [2,5,5,7,3,-5,6]
# [-5, 2, 3, 5, 5, 6, 7]
bubble_sort(arr)
print(arr)

File: Sorting/Bubble Sort/bubble-sort.cpp
#include <bits/stdc++.h>
using namespace std;

void bubbleSort(int arr[], int n)
{
  for (int i = 0; i < n - 1; i++)
  {
    bool swapped = false;
    for (int j = i + 1; j < n; j++)
    {
      if (arr[i] > arr[j])
      {
        swap(arr[i], arr[j]);
        swapped = true;
      }
    }
    if (!swapped)
      return;
  }
}

main()
{
  int n;
  cin >> n;
  int arr[n];
  for (int i = 0; i < n; i++)
    cin >> arr[i];

  bubbleSort(arr, n);

  for (auto x : arr)
    cout << x << " ";
}